spec/schedule_spec.cr

```diff
@@ -155,8 +142,18 @@ describe Schedule do
   end
 
   context ".calculate_interval" do
-    it "provide time for corresponding interval" do
-      Schedule.calculate_interval(:minute).should eq 1.minute
+    it "should return 30 seconds when scheduled at 30th second of a minute" do
+      time = Time.new(2017, 10, 15, 15, 0, 30)
+      Timecop.freeze(time)
+
+      (Schedule.calculate_interval(:minute) < 30.seconds).should be_true
```

Comment:
I am not happy about this spec. It seems the time reduced by 0.00001 second by the time this gets executed. I guess its a bug in `timecop` perhaps or I am doing something wrong.

Reply:
I think this happens because of the [implementation of at_end_of_* methods](https://github.com/crystal-lang/crystal/blob/e2a1389e8165fb097c785524337d7bb7b550a086/src/time.cr#L479). Maybe we should add the remaining time.

Reply:
Rounding it to the second might work. Assuming that the test won't be made to run on a machine that is like damn damn slow :joy: I will do that.